Cubical/Foundations/Pointed/Base.agda

```diff
@@ -0,0 +1,11 @@
+{-# OPTIONS --cubical --safe #-}
+module Cubical.Foundations.Pointed.Base where
+
+open import Cubical.Foundations.Prelude
+
+record Pointed ℓ : Type (ℓ-suc ℓ) where
```

Comment:
Pointed types are already defined in two places in the library:
https://github.com/agda/cubical/blob/master/Cubical/Data/HomotopyGroup/Base.agda#L13
https://github.com/agda/cubical/blob/master/Cubical/Foundations/SIP.agda#L413

It's much better to have them in a separate file, but is it really worth using a record and not just a Sigma? If we define them with a Sigma then the SIP proof will just work as it is. You can still define `typ` and `pt` instead of the normal Sigma projections.

Once this is done could you please remove the duplicate definition in HomotopyGroups? The one in SIP can stay for now as we are working on that file and will refactor later

Reply:
By "SIP proof" I mean the fact that isomorphic pointed types are equal. It's not yet stated in the SIP file, but should be easy to add. Maybe this could be used in some places in your code here (like for `isHomogeneousPath`).

Reply:
Yes, I agree now that this should just be a sigma. My initial rationale was that defining it as a record would force one to use `typ` and `pt` instead of `fst` and `snd`, since using the latter makes code involving pointed types far less readable, from my experience. But having `Pointed ℓ` be definitionally a sigma is definitely more useful.

I had already removed the duplicate definition in HomotopyGroups, but I'm not sure what you have in mind when you suggest to use SIP to prove `isHomogenousPath`.

Reply:
Great! I'm not sure what I had in mind either, so marking this as resolved :-)